src/classes/RD2_ScheduleService.cls

```diff
@@ -1031,6 +1031,16 @@ public without sharing class RD2_ScheduleService {
     */
     public class PauseScheduleHandler {
 
+        public RecurringDonationSchedule__c createPauseSchedule(String statusReason, Date startDate, Date endDate, Id rdId) {
```

Comment:
Should the code in `RD2_PauseForm_CTRL.buildPauseSchedule()` be refactored to avoid the duplication of logic here?